components/registration/processRegistration.js

```diff
@@ -1,10 +1,39 @@
+function showOnlyDisplayInListAttributes(headers, attributesContainer) {
+    const attributeHeaders = headers.slice(7);
+    attributeHeaders
+        .forEach(attributeHeader => {
+            const foundAttributeContainer = attributesContainer.find(attributeContainer => (attributeContainer.trackedEntityAttribute && attributeContainer.trackedEntityAttribute.id)  === attributeHeader.name);
+            if (foundAttributeContainer && !foundAttributeContainer.displayInList) {
+                attributeHeader.hideInList = true;
+            }
+        });
+    return headers;
+};
+
+
 function verifyUniqueSearchGroup(searchGroup, { useProgramSearchScope, SearchGroupService, tetSearchConfig, attributesById, program, trackedEntityType, orgUnit }){
     var promise;
     if(useProgramSearchScope){
         var tetSearchGroup = SearchGroupService.findTetSearchGroup(searchGroup, tetSearchConfig);
-        promise = SearchGroupService.programScopeSearch(searchGroup, tetSearchGroup, program, trackedEntityType, orgUnit);
+        promise = SearchGroupService.programScopeSearch(
+            searchGroup,
+            tetSearchGroup,
+            program,
+            trackedEntityType,
+            orgUnit,
+            undefined,
+            undefined,
+            showOnlyDisplayInListAttributes,
+        );
     }else{
-        promise = SearchGroupService.tetScopeSearch(searchGroup, trackedEntityType, orgUnit);
+        promise = SearchGroupService.tetScopeSearch(
```

Comment:
Formatting changes is great to do as a specific PR, not tangled with the hide field no delete!